### PR TITLE
Add var handle support for boolean and MemoryAddress carrier

### PR DESCRIPTION
**Pull request by mcimadamore**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * [Jorn Vernee](https://openjdk.java.net/census#jvernee) (@JornVernee - Committer) ⚠️ Review applies to a9f0e59d26e51b46e92835648982f5d887abe47a
 * [Paul Sandoz](https://openjdk.java.net/census#psandoz) (@PaulSandoz - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/564/head:pull/564` \
`$ git checkout pull/564`

Update a local copy of the PR: \
`$ git checkout pull/564` \
`$ git pull https://git.openjdk.java.net/panama-foreign pull/564/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 564`

View PR using the GUI difftool: \
`$ git pr show -t 564`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-foreign/pull/564.diff">https://git.openjdk.java.net/panama-foreign/pull/564.diff</a>

</details>

### CODE REVIEW

File: src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryCopy.java [hunk 1]
```diff
@@ -23,10 +23,12 @@
 
 package jdk.incubator.foreign;
 
+import java.nio.ByteBuffer;
```
Comment (leerho) [inline src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryCopy.java:26]: I can't see where ByteBuffer is being used.

**Comment by mcimadamore** [comment]
This change is now ready for review - I did some minor cleanups and added a test for boolean downcalls, which revealed a couple of missing changes in the linker machinery.

File: src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/LayoutPath.java [hunk 2]
```diff
@@ -25,8 +25,10 @@
  */
 package jdk.internal.foreign;
 
+import jdk.incubator.foreign.MemoryAddress;
 import jdk.incubator.foreign.MemoryHandles;
 import jdk.incubator.foreign.MemoryLayout;
+import jdk.incubator.foreign.MemoryLayouts;
```
Comment (JornVernee) [inline src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/LayoutPath.java:31]: The new imports seem to be unused
```suggestion
import jdk.incubator.foreign.MemoryHandles;
import jdk.incubator.foreign.MemoryLayout;
```

**Review by JornVernee — APPROVED** [review]

**Review by PaulSandoz — APPROVED** [review]

**Pull request closed without merging**